lib/CL/pocl_threads_c.h

```diff
@@ -0,0 +1,154 @@
+/* OpenCL runtime library: utility functions for thread operations
```

Comment:
Implemented using pthreads. Maybe the file name should reflect it as well. It's not using the C standard but the POSIX threads standard, if we are strict here. pocl_threads_posix.h?